File: src/perfait/scripts/perfait_scripts/perfait/perfait.cs
using System;
using System.Diagnostics;

namespace perfait {
  public class Perfait {
    public delegate void ProcessEvent();

    static public double Measure(ProcessEvent onProcess){
      var stopwatch = new Stopwatch();
      stopwatch.Start();
      onProcess();
      stopwatch.Stop();
      return stopwatch.Elapsed.TotalSeconds;
    }
  }
}

File: src/perfait/scripts/perfait_scripts/perfait/perfait.js
class Stopwatch {
  static #__get_time(){
    var nowTime = (new Date()).getTime();
    return nowTime / 1000 + ((nowTime % 1000) / 1000);
  }

  #__StartTime;

  constructor(){
    this.start();
  }

  start(){
    this.#__StartTime = Stopwatch.#__get_time();
  }

  stop(){
    return Stopwatch.#__get_time() - this.#__StartTime;
  }
}

class Perfait {
  static measure(onProcess){
    var stopwatch = new Stopwatch();
    onProcess();
    return stopwatch.stop();
  }
}

module.exports = {
  Stopwatch: Stopwatch,
  Perfait: Perfait,
}
